Runtime/Extensions/UnityEngine/SceneExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿using UnityEngine.SceneManagement;
+
+namespace StansAssets.Foundation.Extensions
+{
+    /// <summary>
+    /// Unity `Scene` extension methods.
+    /// </summary>
+    public static class SceneExtensions 
+    {
+        /// <summary>
+        /// Returns the component of Type `type` in the located on scene root GameObject or any of its children using depth first search.
+        /// A component is returned only if it is found on an active GameObject.
+        /// </summary>
+        /// <param name="scene">Scene to operate with.</param>
+        /// <typeparam name="T">Type of the component.</typeparam>
+        /// <returns>A component of the matching type, if found.</returns>
+        public static T GetComponent<T>(this Scene scene) {
```

Comment:
Maybe a good idea to use the same naming as Unity has, I mean in the case when we looking for a Component in Scene roots and their children - this method can be called **GetComponentInChildren**?

Reply:
Also as Codacy noticed: would be better to add a constraint for T parameter:
`public static T GetComponent<T>(this Scene scene) where T : class`
Because otherwise possible to pass value type there.

Reply:
Thanks for the very valid point!